client/src/components/HistoryViewer/HistoryViewerVersion.js

```diff
@@ -36,44 +42,51 @@ class HistoryViewerVersion extends Component {
     return `${member.FirstName || ''} ${member.Surname || ''}`;
   }
 
+  /**
+   * Return a string of HTML class names for the row (list item) element
+   *
+   * @returns {string}
+   */
+  getClassNames() {
+    const { extraClass, isActive, compare, compare: { compareFrom, compareTo } } = this.props;
+    const defaultClasses = {
+      'history-viewer__row': true,
+      'history-viewer__row--current': isActive,
+      'history-viewer__row--comparison-selected': compare && !(compareFrom && compareTo),
+    };
+    return classNames(defaultClasses, extraClass);
+  }
+
   /**
    * When clicking on a version, render the detail view for it via a Redux action dispatch
+   * passed through via a closure prop (onSelect)
    */
   handleClick() {
-    const { onSelect, version, isActive, compare, onCompareSelect } = this.props;
+    const { onSelect, version, isActive, compare } = this.props;
 
     // If the clear button is shown, don't do anything when clicking on the row
     if (isActive) {
       return false;
     }
 
-    if (compare) {
-      onCompareSelect(version.Version, !compare.versionFrom);
-    } else {
-      onSelect(version.Version);
-    }
-
+    onSelect(version, compare);
     return false;
   }
 
+  handleCompare() {
+    const { onCompareMode, version } = this.props;
+    onCompareMode(version);
+  }
+
   /**
    * When closing the version, return back to the list view via Redux action dispatch
    */
   handleClose() {
-    const { onSelect, compare, onCompareSelect, version } = this.props;
-    if (compare) {
-      onCompareSelect(0, version.Version === compare.versionFrom);
-    } else {
-      onSelect(0);
+    const { onSelect, version, compare, compare: { versionFrom } } = this.props;
+    if (versionFrom && versionFrom.Version === version.Version) {
+      delete compare.versionFrom;
```

Comment:
Question: are we storing the versionFrom and versionTo in the redux store? If yes, should this be dispatching a redux action to clear them rather than modifying the props? I notice we have no `mapStateToProps` in this component now (it's removed below). If not, should we be?

Reply:
This component is now responsible for displaying data, not managing it's own state.
It is still connected to the store, but only to be able to dispatch events, not to read state and inflate itself. Data is send down from above as props.
The purpose of this action here is not to update the state, but rather trigger the correct action when dispatching, as checks are made against the data to evaluate the dispatch to make.

c.f. `mapStateToProps->onSelect`

Reply:
Ok, so I'm assuming here that the prop you say is passed down originally comes from the redux store, but this particular piece of code is modifying its own prop directly (which is sort of state management). Does the parent component's `onSelect` need to handle the removal of versionFrom from the redux store, which would in turn pass the falsy value down to this component and update its prop automatically?

Reply:
Ok, thanks for clarifying